ocaml/idl/datamodel.ml

```diff
@@ -8638,6 +8649,47 @@ let vgpu_type =
 		]
 	()
 
+(** PVS_farm
+ * http://www.uk.xensource.com/~robho/pvs-direct/#external-interfaces-xenapi-pvs_farm-new-class
+ *)
+
+
+let pvs_farm =
+  let lifecycle = [Published, rel_dundee_plus, ""] in
+  let refset x  = Set(Ref x) in
+    create_obj
+    ~name: _pvs_farm
+    ~descr:"machines serving blocks of data for provisioning VMs"
+    ~doccomments:[]
+    ~gen_constructor_destructor:false
+    ~gen_events:true
+    ~in_db:true
+    ~lifecycle:lifecycle
+    ~persist:PersistEverything
+    ~messages:[]
+    (* ~mesages_default_allowed_roles:_R_POOL_OP *)
+    ~in_oss_since:None
+    ~contents:
+    [ uid     _pvs_farm ~lifecycle:lifecycle
+
+    ; field   ~qualifier:StaticRO ~lifecycle:lifecycle
+              ~ty:String "name"
+              "Name of the PVS farm. Must match name configured in PVS"
+
+    ; field   ~qualifier:StaticRO ~lifecycle:lifecycle
+              ~ty:(Ref _sr) "cache_sr"
+              "The SR used by PVS proxy for the cache"
+
+    ; field   ~qualifier:StaticRO ~lifecycle:lifecycle
```

Comment:
I think that this on and the next should be DynamicRO (though in practice it doesn't really matter, because the class does not have an auto-generated constructor).